docs/docsite/rst/dev_guide/platforms/vmware_guidelines.rst

```diff
@@ -0,0 +1,180 @@
+    vcenter_hostname: vcenter.test
+    vmware_validate_certs: false
+    esxi1_username: root
+    esxi1_hostname: esxi1.test
+    esxi1_password: root
+    esxi2_username: root
+    esxi2_hostname: test2.test
+    esxi2_password: root
+
+If your test-infrastructure is behinf a HTTP proxy, you will need the following extra pull-requests:
+
+- ansible-test: vcenter behind a HTTP proxy <https://github.com/ansible/ansible/pull/58208>
+- pyvmomi Proxy support <https://github.com/vmware/pyvmomi/pull/799>
+- VMware: add support for HTTP proxy in connection API <https://github.com/ansible/ansible/pull/52936>
+
+The specify the location of the proxy server with the two extra keys:
+
+.. code-block:: ini
+
+    vmware_proxy_host: esxi1-gw.ws.testing.ansible.com
+    vmware_proxy_port: 11153
+
+In addition, you may need to adjust the variables of the following file to match the configuration of your lab:
+:file:`test/integration/targets/prepare_vmware_tests/vars/real_lab.yml`. If you use `vmware-on-libvirt <https://github.com/goneri/vmware-on-libvirt>` to prepare you lab, you don't have anything to change.
+
+Run the test-suite
+------------------
+
+Once your configuration is ready, you can trigger a run with the following command:
+
+.. code-block:: shell
+
+    source hacking/env-setup
+    VMWARE_TEST_PLATFORM=static ansible-test integration --python 3.7 vmware_host_firewall_manager
+
+``vmware_host_firewall_manager`` is the name of the module to test.
+
+Unit-test
+=========
+
+The VMware modules come with a limited unit-test coverage, you can run the test-suite with the
+following commands:
+
+.. code-block:: shell
+
+    source hacking/env-setup
+    ansible-test units --tox --python 3.7 '.*vmware.*'
+
+Code style and best practice
+============================
+
+datacenter argument with ESXi
+-----------------------------
+
+``datacenter`` parameter should not use ``ha-datacenter`` by default. This is because the user may
+not realize that Ansible silently targets the wrong data center.
+
+esxi_hostname should not be mandatory
+-------------------------------------
+
+Depending upon the functionality provided by ESXi or vCenter, some modules can seamlessly work with both. In this case,
+``esxi_hostname`` parameter should be optional.
+
+.. code-block:: python
+
+    if self.is_vcenter():
+        esxi_hostname = module.params.get('esxi_hostname')
+        if not esxi_hostname:
+            self.module.fail_json("esxi_hostname parameter is mandatory")
+        self.host = self.get_all_host_objs(cluster_name=cluster_name, esxi_host_name=esxi_hostname)[0]
+    else:
+        self.host = find_obj(self.content, [vim.HostSystem], None)
+    if self.host is None:
+        self.module.fail_json(msg="Failed to find host system.")
+
+Typographic convention
+======================
+
+Nomenclature
+------------
+
+We try to enforce the following rules in our documentation:
+
+- VMware, not VMWare or vmware
+- ESXi, not esxi
+- VCenter, not vcenter
```

Comment:
Based on the comments above, I thought the correct capitalization was `vCenter` . . . if I'm correct:

```suggestion
- vCenter, not vcenter
```

Reply:
oh, thanks @acozine for your review. You're right, it looks so much better.